Minor fixes for Intel C/C++ compilers

amadio opened the pull request.

amadio committed:
Do not warn users about "broken" FMA4 and/or XOP

Even AMD processors use FMA3 these days, so on most CPUs
FMA4 is "broken". CPUs supporting XOP and FMA4 are mostly
obsolete.

amadio committed:
Remove special case of MayAlias<T> for Intel C++ compiler

Also, disable warnings about attributes, same as for GCC.

bernhardmgruber commented:
I am fine with the commit streamlining the `MayAlias` implementation.

For the other two commits dropping the obsolete instruction sets: could you please move those to a separate PR and also inspect the C++ source code whether there is any special handling of those instruction sets if they are available? Thx!

amadio committed:
Do not compile for XOP+FMA4 if compiler cannot do it

Fixes the warning below when building with the Intel compiler:

CMake Warning at cmake/VcMacros.cmake:503 (message):
  The following unknown targets where listed in the ONLY list of
  vc_compile_for_all_implementations: 'SSE+XOP+FMA4;AVX+XOP+FMA4;AVX+XOP+FMA'
Call Stack (most recent call first):
  CMakeLists.txt:121 (vc_compile_for_all_implementations)

amadio commented:
Ok, rather than remove, I just conditionally enable the compilation of the XOP+FMA4 targets. This is enough to get rid of the warnings with the Intel compiler, which cannot do XOP+FMA4. I think since all changes here are related to improving things with the Intel compiler, it's ok to keep them together in this pull request, no?

amadio commented:
This actually caused a warning with MSVC, so I was waiting to merge until I got that fixed... Ok I can add the fix afterwards.
